CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc

```diff
@@ -1606,52 +1713,54 @@ void EcalTPGParamBuilder::analyze(const edm::Event& evt, const edm::EventSetup&
   // Compute weights section //
   /////////////////////////////
 
-  const int NWEIGROUPS = 2;
-  std::vector<unsigned int> weights[NWEIGROUPS];
-
-  shapeEB_.setEventSetup(evtSetup);
-  shapeEE_.setEventSetup(evtSetup);
-  weights[0] = computeWeights(shapeEB_, hshapeEB);
-  weights[1] = computeWeights(shapeEE_, hshapeEE);
-
-  map<EcalLogicID, FEConfigWeightGroupDat> dataset;
-
-  for (int igrp = 0; igrp < NWEIGROUPS; igrp++) {
-    if (weights[igrp].size() == 5) {
-      if (writeToFiles_) {
-        (*out_file_) << std::endl;
-        (*out_file_) << "WEIGHT " << igrp << endl;
-        for (unsigned int sample = 0; sample < 5; sample++)
-          (*out_file_) << "0x" << hex << weights[igrp][sample] << " ";
-        (*out_file_) << std::endl;
-        (*out_file_) << std::endl;
-      }
-      if (writeToDB_) {
-        ss << "going to write the weights for groupe:" << igrp << "\n";
-        FEConfigWeightGroupDat gut;
-        gut.setWeightGroupId(igrp);
-        //PP WARNING: weights order is reverted when stored in the DB
-        gut.setWeight0(weights[igrp][4]);
-        gut.setWeight1(weights[igrp][3] +
-                       0x80);  //0x80 to identify the max of the pulse in the FENIX (doesn't exist in emulator)
-        gut.setWeight2(weights[igrp][2]);
-        gut.setWeight3(weights[igrp][1]);
-        gut.setWeight4(weights[igrp][0]);
-        EcalLogicID ecid = EcalLogicID("DUMMY", igrp, igrp);  //1 dummy ID per group
-        // Fill the dataset
-        dataset[ecid] = gut;
+  std::map<int, int> weights_map_even;
+  std::map<int, int> weights_map_odd;
+
+  if (weight_even_computeFromShape_) {
```

Comment:
should there be some fallback if this is `false`?

Reply:
There is one at L1798. In that case the weights are read from file